### DISCUSSION TITLE
Can't modify any server configuration

### DISCUSSION THREAD

**Discussion by tomlawesome** [immense/Remotely#377] [General]
Hi, 

I'm really struggling to resolve issues with known proxies/CORS. After many hours of trying to debug this, I believe that CORS/proxies are the problem, because it all works absolutely fine on a local ip direct to the server. I also googled the errors, and there was a lot of discussion around CSRF tokens, origins, CORS etc. 

I just can't seem to get my configuration correct. 

My setup is:

Cloudflare proxy --> OPNsense --> Server (VM on Proxmox) --> Traefik (docker) --> Remotely (docker). 

This is the only error I'm getting:

```
warn: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[35]
      No XML encryptor configured. Key {348adfb3-fcb9-4a84-81a2-89491c548348} may be persisted to storage in unencrypted form.
warn: Microsoft.AspNetCore.HttpsPolicy.HttpsRedirectionMiddleware[3]
      Failed to determine the https port for redirect.
fail: Microsoft.AspNetCore.Antiforgery.DefaultAntiforgery[7]
      An exception was thrown while deserializing the token.
      Microsoft.AspNetCore.Antiforgery.AntiforgeryValidationException: The antiforgery token could not be decrypted.
       ---> System.Security.Cryptography.CryptographicException: The key {7af85ed0-07bd-47ba-9e6f-02fe1dcd3630} was not found in the key ring.
         at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingBasedDataProtector.UnprotectCore(Byte[] protectedData, Boolean allowOperationsOnRevokedKeys, UnprotectStatus& status)
         at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingBasedDataProtector.DangerousUnprotect(Byte[] protectedData, Boolean ignoreRevocationErrors, Boolean& requiresMigration, Boolean& wasRevoked)
         at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingBasedDataProtector.Unprotect(Byte[] protectedData)
         at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgeryTokenSerializer.Deserialize(String serializedToken)
         --- End of inner exception stack trace ---
         at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgeryTokenSerializer.Deserialize(String serializedToken)
         at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgery.GetCookieTokenDoesNotThrow(HttpContext httpContext)

```
I've added the CF proxy ipv4 address', as well as my router IP, traefik's container IP, the docker gateway IP. I've added sub.domain.net to CORS in Remotely config. Traefik includes headers: 

```
    middlewares-secure-headers:
      headers:
        accessControlAllowMethods:
          - GET
          - OPTIONS
          - PUT
          - POST
          - DELETE
          - PATCH
        accessControlAllowOriginList:
          - 'https://sub.domain.net'

```
But still, despite all that, if I try to edit any settings by the FQDN it fails to do anything. 

![image](https://user-images.githubusercontent.com/76453276/139112235-b563194c-3dd2-4b41-9efa-b03fbbe8560c.png)

**Comment by scockman** [comment]
I am having a similar issue but using Cloudflare Tunnel.

**Comment by Purger-BB** [comment]
I had the same issue using cloudflare.
What I did to resolve this is creating a page rule for remotely subdomain (should work for a tunnel too).
Under page rules, create new rule and add this rules:

Cache level: Bypass
Automatic HTTPS Rewrites: Off
Disable Apps
Disable Performance

Let it sit for a few minutes, and try again, it should work.

**Reply by mendesit** [reply]
Good night @Purger-BB, where can i setup this to Remotely work behind cloudflare tunnel?
